resources/qml/device-verification/DeviceVerification.qml

```diff
@@ -0,0 +1,634 @@
+import QtQuick 2.3
+import QtQuick.Controls 2.10
+import QtQuick.Window 2.2
+import QtQuick.Layouts 1.10
+
+import im.nheko 1.0
+
+ApplicationWindow {
+	title: stack.currentItem.title
+	id: dialog
+
+	flags: Qt.Dialog
+
+	palette: colors
+
+	height: stack.implicitHeight
+	width: stack.implicitWidth
+	StackView {
+		id: stack
+		initialItem: flow.sender == true?newVerificationRequest:acceptNewVerificationRequest
+		implicitWidth: currentItem.implicitWidth
+		implicitHeight: currentItem.implicitHeight
+	}
+
+	property var flow
+	property bool isRequest
+	property var tran_id
+
+	Connections {
+		target: flow
+		onVerificationCanceled: stack.replace(partnerAborted)
+		onTimedout: stack.replace(timedout)
+		onDeviceVerified: stack.replace(verificationSuccess)
+
+		onVerificationRequestAccepted: switch(method) {
+			case DeviceVerificationFlow.Decimal: stack.replace(digitVerification); break;
+			case DeviceVerificationFlow.Emoji: stack.replace(emojiVerification); break;
+		}
+
+		onRefreshProfile: {
+			deviceVerificationList.updateProfile(flow.userId);
+		}
+	}
+
+	Component {
```

Comment:
Can you put these Components into their own file and then use one of the approaches listed [here](https://forum.qt.io/topic/53056/pushing-qml-files-onto-stackview/7) to do the stack.replace? This file is getting a bit big.